econml/bootstrap.py

```diff
@@ -5,6 +5,254 @@
 import numpy as np
 from joblib import Parallel, delayed
 from sklearn.base import clone
+from scipy.stats import norm
+from collections import OrderedDict
+import pandas as pd
+
+
+class BootstrapInferenceResults:
+    """
+    Results class for bootstrap inference.
+
+    Parameters
+    ----------
+    pred_dist : array-like, shape (b, m, d_y, d_t) or (b, m, d_y)
+        the raw predictions of the metric using b times bootstrap.
+        Note that when Y or T is a vector rather than a 2-dimensional array,
+        the corresponding singleton dimensions should be collapsed
+    kind : 'percentile' or 'pivot'
+        Whether to use percentile or pivot-based intervals
+    d_t: int
+        Number of treatments
+    d_y: int
+        Number of outputs
+    inf_type: string
+        The type of inference result.
+        It could be either 'effect', 'coefficient' or 'intercept'.
+    fname_transformer: None or predefined function
+        The transform function to get the corresponding feature names from featurizer
+    """
+
+    def __init__(self, pred_dist, kind, d_y, d_t, inf_type, fname_transformer):
+        self.pred_dist = pred_dist
```

Comment:
Wouldnt' it suffice if you check whether kind='pivot' or 'percentile' and simply changed the pred_dist here? i.e. if pivot then set `self.pred_dist = 2*point_estimate - pred_dist`. Then you don't need to check for pivot or percentile in each of the calculations. 

Also it seems that similar to inference results, point_estimate should just be another input parameter provided by the cate estimator class as input to the inference results and not computed here as a function of pred_dist.

Reply:
Yes, this is fixed in the latest update.  There is now an `EmpiricalInferenceResults` subclass that takes the distribution as an input and uses it as-is; the calling code in bootstrap is now responsible for doing the pivot logic itself before passing the array in.